src/main/java/org/jabref/model/entry/BibEntry.java

```diff
@@ -812,6 +812,15 @@ private void invalidateFieldCache(String fieldName) {
             Optional<String> citeKey = getCiteKeyOptional();
             latexFreeFields.put(name, citeKey.get());
             return citeKey;
+        } else if (TYPE_HEADER.equals(name)) {
+            Optional<EntryType> entryType = EntryTypes.getType(getType(), BibDatabaseMode.BIBLATEX);
+            if (entryType.isPresent()) {
+                String entryName = entryType.get().getName();
+                latexFreeFields.put(name, entryName);
+                return Optional.of(entryName);
+            } else {
+                return Optional.empty();
```

Comment:
In the case when the entry is unknown, you can nonetheless return the value of `getType` as in:
https://github.com/JabRef/jabref/blob/26cd572d4e0587ddf763edc3f9be4caafb1844e6/src/main/java/org/jabref/model/entry/BibEntry.java#L137-L142